### PR TITLE
SW-5280 Update with BE types for new species enum sets

### PR DESCRIPTION
**Pull request by nickgraz**
- New species fields `plant_material_sourcing_methods` and `successional_groups`
- Removes mock data for those fields since it will come back with the API
- Removes hard coded enum values and uses server values

Depends on this `terraware-server` PR https://github.com/terraware/terraware-server/pull/1892

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2557?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2557** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2557?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2551** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2551?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2550** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2550?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2549** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2549?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/types/Species.ts [hunk 1]
```diff
@@ -46,58 +46,27 @@ export const mockSpeciesNewFieldsData: Partial<Species> = {
...
 
-export type SeedStorageBehavior =
-  | 'Intermediate'
-  | 'Likely Intermediate'
-  | 'Likely Orthodox'
-  | 'Likely Recalcitrant'
-  | 'Orthodox'
-  | 'Recalcitrant'
-  | 'Unknown';
+export type SeedStorageBehavior = NonUndefined<
```
Comment (constanzauanini) [inline src/types/Species.ts:67]: I think you can use Required<> here, like in https://github.com/terraware/terraware-web/blob/main/src/types/Accession.ts#L9

Comment (nickgraz) [reply src/types/Species.ts:67]: That didn't seem to work unfortunately:

**Review by joshbuchea — APPROVED** [review]
Nice!

**Pull request merged**